packages/components/src/components/languageSupport/squiggle.grammar

```diff
@@ -0,0 +1,104 @@
+
+@precedence {
+    deref,
+    call,
+    unary @right,
+    control @left,
+    exp @left,
+    times @left,
+    plus @left,
+    else @right,
+    rel @left,
+    and @left,
+    or @left
+}
+
+@top Program {
+    (Binding newline | newline)* (expression newline*)?
+}
+
+@top Statement {
+    expression
+}
+
+commaSep<content> {
+    "" | content ("," content)*
+}
+
+Binding { VariableName { identifier } "=" expression }
+
+LambdaArgs {
+    "" | LambdaParameter { identifier } ("," LambdaParameter { identifier })*
+}
+
+AccessIdentifier {
+    Field { identifier } (!deref "." Field { identifier })*
+}
+
+expression {
+    String
+    | Boolean
+    | Number
+    | RecordExpr { "{" commaSep<Entry { Field {identifier} ":" expression }> "}" }
+    | LambdaExpr { "{" "|" LambdaArgs "|" expression "}" }
+    | IfExpr { if expression then expression !else else expression }
+    | ParenExpr { "(" expression ")" }
+    | AccessExpr { AccessIdentifier }
+    | CallExpr { expression !call "(" commaSep<Argument { expression }> ")" }
+    | ArrayExpr { "[" commaSep<expression> "]" }
+    | Void { !deref "()" }
+    | UnaryExpr { !unary ArithOp { "-" } expression }
+    | LogicExpr { 
+          expression !or     LogicOp    { "||" }  expression
+        | expression !and    LogicOp    { "&&" }  expression
+        | expression !rel    LogicOp    { ">" }   expression
+        | expression !rel    LogicOp    { "<" }   expression
+        | expression !rel    LogicOp    { "<=" }  expression
+        | expression !rel    LogicOp    { ">=" }  expression
+        | expression !rel    LogicOp    { "==" }  expression
+    }
+    | ControlExpr { 
+          expression !times  ControlOp  { "|>" }  expression
+    }
+    | ArithExpr { 
+          expression !times  ArithOp    { "*" }   expression
+        | expression !times  ArithOp    { "/" }   expression
+        | expression !exp    ArithOp    { "^" }   expression
+        | expression !plus   ArithOp    { "+" }   expression
+        | expression !plus   ArithOp    { "-" }   expression
+    }
+}
+
+
+
+Boolean { @specialize[@name="Boolean"]<identifier, "true" | "false"> }
+
+kw<term> { @specialize[@name={term}]<identifier, term> }
+if { kw<"if"> }
+then { kw<"then"> }
+else { kw<"else"> }
+
+@skip { spaces | Comment }
+
+@tokens {
+
+    spaces[@export] { $[ ]+ }
+    newline[@export] { $[\n] }
+    Comment { "#" ![\n]* }
```

Comment:
Lol, I'm not sure if I like this, but squiggle also supports C style comments:
https://github.com/quantified-uncertainty/squiggle/blob/3012aef683ab454b947d00e1c49d3a8bd178a57d/packages/vscode-ext/syntaxes/squiggle.tmLanguage.yaml#L61-L70

Reply:
Same goes for block comments `/* */`

Reply:
Oooh. Will be patching that and multiline lambdas in. Thank you for testing!

Reply:
Alas there are still weird things in peggy, like its not tolerating `b = { || () } |> { |f| () }` or even `true == false == true`

Reply:
More peggy fun:
This parses
```
f() = {
a = ()
;;;;;;;
12
}
```
This does not
```
f() = {
a = ()
  ;;;;;;;
12
}
```